pkg/client/observatorium/api_mock.go

```diff
@@ -26,13 +28,13 @@ func (t *httpAPIMock) QueryExemplars(ctx context.Context, query string, startTim
 }
 
 // performs a query for the kafka metrics.
-func (t *httpAPIMock) Query(ctx context.Context, query string, ts time.Time) (pModel.Value, pV1.Warnings, error) {
+func (t *httpAPIMock) Query(ctx context.Context, query string, ts time.Time, opts ...pV1.Option) (pModel.Value, pV1.Warnings, error) {
```

Comment:
cc @pb82 to review the changes in this PR.

I think with the changes we've done we are ok. We don't need to take the options into account at all due to this is part of the `httpAPIMock` type which is a mock and we don't need to process them at all.

Independently on that maybe we should consider if it would be interesting to process the options when using the real client in a future PR if we deem it would be useful. cc @pb82

Reply:
thanks for updating this @miguelsorianod 
Looks like the only available option at the moment is a timeout: https://github.com/prometheus/client_golang/blob/83d56b1144a0c2eb10d399e7abbae3333bebc463/api/prometheus/v1/api.go#L820

But regardless, it can be useful in the future.